src/layerPart.cpp

```diff
@@ -65,22 +65,60 @@ void createLayerWithParts(const Settings& settings, SliceLayer& storageLayer, Sl
         result = layer->polygons_.splitIntoParts(union_layers || union_all_remove_holes);
     }
 
-    for (auto& part : result)
+    for (auto& main_part : result)
     {
-        storageLayer.parts.emplace_back();
-        if (part.empty())
+        std::vector<std::pair<SliceLayerPart::WallExposedType, std::vector<SingleShape>>> parts_by_type = {
+            { SliceLayerPart::WallExposedType::BOTTOM_0, bottom_parts.splitIntoParts() },
+            { SliceLayerPart::WallExposedType::TOP, top_parts.difference(bottom_parts).splitIntoParts() },
+            { SliceLayerPart::WallExposedType::SIDE_ONLY, main_part.difference(bottom_parts).difference(top_parts).splitIntoParts() },
+        };
```

Comment:
Maybe a `std::map` would make things easier ? unless you need an ordering

Reply:
... I'm not sure why I didn't see that myself (probably because I didn't conceptually see it as a mapping, given what I do next with it).

I don't think it makes things easier per-se, but it'd certainly make the type easier to read.

Reply:
That is right, it is fundamentally not used as a key-value container, but I tend to use more std::map because it simplifies the type. This is mostly a personal preference.

Reply:
I ended up using both a map for this and another one to replace the function/switch, so I think this PR is now an atlas :-p